pgcli/pgexecute.py

```diff
@@ -233,14 +233,22 @@ def connect(self, database=None, user=None, password=None, host=None,
             'dsn': dsn,
         }
         new_params.update(kwargs)
+
+        if new_params['dsn']:
```

Comment:
Looks like I am too late for this, but you want to keep that `make_dsn` call. You do not want to re-format DSN by hand like this PR does. Reason: https://github.com/dbcli/pgcli/issues/1043 (fixed by https://github.com/dbcli/pgcli/pull/1045/files).

Reply:
Doh! I'll take a look at it when I land. Sorry about being trigger happy with the merge. 

How come the tests passed? :thinking:

Reply:
We do not have a test case for password with space. Probably should add one.

Reply:
I just tried to connect to a database with a space in the password it seems to connect just fine. What am I missing?

Reply:
You'd only get into this branch when connecting with a `dsn` parameter. So for example you have this service defined:

```
$ cat ~/.pg_service.conf                                                                                                                                                                                               [service_postgres]
dbname=postgres
host=localhost
```

and you connect like this:

`pgcli service=service_postgres`

and you have a password with space in it. It will break there.